easybuild/tools/filetools.py

```diff
@@ -2270,6 +2270,9 @@ def copy_file(path, target_path, force_in_dry_run=False):
     :param target_path: path to copy the file to
     :param force_in_dry_run: force copying of file during dry run
     """
+    # NOTE: 'exists' will return False if 'path' is a broken symlink
+    if not os.path.exists(path) and not os.path.islink(path):
+        raise EasyBuildError("Could not copy '%s' it does not exist!" % path)
```

Comment:
Hmm, do we also want this to happen during dry run mode (`eb -x`)?

I'd say probably not, because there it's very common to not have the files actually exist, and it would trigger lots of failures when using `eb -x` (see the output of `eb -x Eigen-3.2.8.eb` for example, this change makes it trip during `install_step`)

So, an extra guard is required here:

```python
if not os.path.exists(path) and not os.path.islink(path):
    if force_in_dry_run:
        raise EasyBuildError("Could not copy '%s' it does not exist!", path)
    else:
        _log.debug("Ignoring non-existing file in copy_file because of dry run mode: %s", path)
```

The test should be enhanced accordingly, it should check that the error is not raised when in dry run mode (you can use `update_build_option('extended_dry_run', True)`, and also test with `force_in_dry_run=True`).

It's also better to use lazy logging: `, path` rather than `% path`